utilityTypes.ts

```
@@ -17,13 +17,34 @@
     // Handle the general case of objects, where keys are either required *or* include undefined as a union
     values<T extends {}>(o: T): keyof T extends never ? never : [T[keyof T], ...T[keyof T][]];
 
-    assign<T extends {}, U extends {}>(target: T, source: U): Omit<T, keyof U> & U;
+    assign<T extends {}, U extends {}>(target: T, source: U): {} extends T ? U : Omit<T, keyof U> & U & {[K in Extract<OptionalKeyof<U>, keyof T>]: Extract<T[K] & U[K], undefined> extends never ? Exclude<T[K] | U[K], undefined> : T[K] | U[K]};
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)